### DISCUSSION TITLE
Use X-Bind: to Set Custom Data- HTML Attributes

### DISCUSSION THREAD

**Discussion by maweitzman** [alpinejs/alpine#2384] [1. Help]
Maybe I'm just missing it somehow, but I've repeatedly tried setting custom `data-` HTML attributes with `x-bind:` with no success. Anyone know how to accomplish this, or is this an Alpine limitation/bug?

```html
<span :data-tippy-content="sample expression to evaluate && 'sample tooltip copy'">
```

**Comment by SimoTod** [comment]
The content of your attribute resolves to a boolean. Is that what you want?

**Reply by maweitzman** [reply]
@SimoTod per Alpine's docs: https://alpinejs.dev/directives/bind#shorthand-conditionals, what I understand my code to read as is:

If "sample expression to evaluate" is true, then apply the value "sample tooltip copy" to the `data-tippy-content` HTML attribute. Am I misunderstanding?

**Reply by SimoTod** [reply]
I think you are right, i don't use that syntax, my bad

**Reply by SimoTod** [reply]
Does tippy support dynamic data attributes. (does it work if you build a page without Alpine and you update the attribute in dev console?)

**Reply by maweitzman** [reply]
Now that I think about it, I've had trouble with this in the past with Tippy, so I think I have to accept that it ain't gonna happen without a ton of extra effort.

**Reply by SimoTod** [reply]
Try https://github.com/ryangjchandler/alpine-tooltip. It should take care of the reactive part for you.
